zowex/src/main.rs

```diff
@@ -726,6 +767,289 @@ fn run_nodejs_command(cmd_line_args: &mut Vec<String>) -> Result<i32, i32> {
+    njs_zowe_path: &'a str,
+    curr_cmd_shell: & CmdShell,
+    script_string: &'a mut String,
+    script_arg_vec: &mut Vec<&'a str>
+) -> String {
+    const SCRIPT_WAIT_MSG: &str = "echo Wait to see the results below ... ";
+    const SCRIPT_PROMPT_MSG_FIXED: &str = "echo Now press ENTER to see your command ";
+
+    if env::consts::OS == "windows"
+    {
+        return form_win_cmd_script_arg_vec(
+            zowe_cmd_args, njs_zowe_path, curr_cmd_shell,
+            SCRIPT_WAIT_MSG, SCRIPT_PROMPT_MSG_FIXED,
+            script_arg_vec
+        );
+    }
+
+    form_bash_cmd_script_arg_vec(
+        zowe_cmd_args, njs_zowe_path, script_string,
+        SCRIPT_WAIT_MSG, SCRIPT_PROMPT_MSG_FIXED,
+        script_arg_vec
+    )
+}
+
+/**
+ * Form the argument vector required to launch deamon enable and disable commands
+ * on Windows. For Windows, each space-separated item on the script's command
+ * line goes into a separate script_arg_vec element.
+ *
+ * @param zowe_cmd_args
+ *      The user-supplied command line arguments to the zowe command.
+ *
+ * @param njs_zowe_path
+ *      Path to our Node.js zowe command.
+ *
+ * @param curr_cmd_shell
+ *      The current command shell under which we are running.
+ *
+ * @param script_wait_msg
+ *      A text message telling the user to wait for our background process.
+ *
+ * @param script_prompt_msg_fixed
+ *      The fixed part of text message telling the user to press ENTER to get a prompt.
+ *
+ * @param script_arg_vec
+ *      An empty vector into which we place command script arguments.
+ *
+ * @returns The command shell program that we will launch in the
+ *          background to run our script.
+ */
+fn form_win_cmd_script_arg_vec<'a>(
+    zowe_cmd_args: &'a [String],
+    njs_zowe_path: &'a str,
+    curr_cmd_shell: & CmdShell,
+    script_wait_msg: &'a str,
+    script_prompt_msg_fixed: &'a str,
+    script_arg_vec: &mut Vec<&'a str>
+) -> String {
+    // add any required newlines to create some space
+    script_arg_vec.push("/C");
+    script_arg_vec.push("echo.");
+    script_arg_vec.push("&&");
+
+    if matches!(curr_cmd_shell, CmdShell::PowerShellDotNet | CmdShell::PowerShellExe) {
+        // PowerShell needs extra newlines before the background process output to create space
+        for _count in [1,2,3,4,5,6] {
+            script_arg_vec.push("echo.");
+            script_arg_vec.push("&&");
+        }
+
+    } else if matches!(curr_cmd_shell, CmdShell::Bash | CmdShell::Sh) {
+        // Bash shell on windows needs a delay and a newline in its spacing
+        for next_arg in "sleep 1 && echo. &&".split_whitespace() {
+            script_arg_vec.push(next_arg);
+        }
+    }
+
+    // add our wait message
+    for next_arg in script_wait_msg.split_whitespace() {
+        script_arg_vec.push(next_arg);
+    }
+    script_arg_vec.push("&&");
+
+    // make script delay so the EXE can exit
+    for next_arg in "ping 127.0.0.1 -n 1 >nul &&".split_whitespace() {
```

Comment:
Why do we need a `ping` command here and how does it make the script delay? Sorry if I'm missing something obvious 😢

Reply:
There is no native "sleep" command on Windows that runs in all Windows command shells. My web searches came up with ping as the best always-available command. The ping takes some time to perform its operation. The more request packets that you ask for, the longer the command takes. I read the each request takes about a second. Only doing one request (-n 1) seemed to reliably delay long enough for me. We could increase the value of the -n flag if we find that more time is needed to allow the EXE to exit at other sites.

Reply:
Thanks for the explanation, Windows seems silly to not have a `sleep` command but this makes sense now.